src/Service/Assembler/AssemblerInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\Assembler;
+
+interface AssemblerInterface
+{
+    public function read($entityData, $entity = null);
+
+    public function write($entity);
+
+    public function writeMultiple(array $entities): array;
```

Comment:
Array is quite vague... can you use typed collections or at least `(Entity ...$entities)` with a splat operator?